docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md

```diff
@@ -0,0 +1,158 @@
+    public
+    returns (uint256)
+  {
+    _tokenIds.increment();
+
+    uint256 newItemId = _tokenIds.current();
+    _mint(player, newItemId);
+    // _setTokenURI(newItemId, tokenURI);
+
+    return newItemId;
+  }
+}
+```
+
+Let's examine this implementation of an NFT as a Game Item. We start by importing to contracts from our node modules. We import Openzeppelin's open source implementation of the ERC721 standard which our NFT contract will inherit from. Our constructor takes the `_name` and `_symbol` arguments for our NFT and passes them on to the constructor of the parent ERC721 implementation. Lastly we implement the `awardItem` function which allows anyone to mint an NFT to a player's wallet address. This function increments the `currentTokenId` and makes use of the `_mint` function of our parent contract.
+
+# Compile & deploy with forge
+
+To compile the NFT contract run:
+
+ ```zsh
+ forge build
+ ``` 
+ 
+ By default the compiler output will be in the `out` directory. To deploy our compiled contract with Forge we have to set environment variables for the RPC endpoint and the private key we want to use to deploy.
+
+Set your environment variables by running:
+
+```zsh
+export RPC_URL=<YOUR-RPC-ENDPOINT>
+export PRIVATE_KEY=<YOUR-PRIVATE-KEY>
+```
+
+Since we are deploying to Fuji testnet, our `RPC_URL` export should be:
+
+```zsh
+export RPC_URL=https://api.avax-test.network/ext/bc/C/rpc
+```
+
+Once set, you can deploy your NFT with Forge by running the command below while adding the relevant constructor arguments of the NFT contract:
+
+```zsh
+forge create NFT --rpc-url=$RPC_URL --private-key=$PRIVATE_KEY --constructor-args <NFT-NAME> <NFT-SYMBOL> 
+```
+
+Upon successful deployment, you will see the deploying wallet's address, the contract's address as well as the transaction hash printed to your terminal.
+
+Here's an example output from an NFT deployment.
+
+```zsh
+[⠔] Compiling...
+No files changed, compilation skipped
+Deployer: 0x8db97c7cece249c2b98bdc0226cc4c2a57bf52fc
+Deployed to: 0x52c84043cd9c865236f11d9fc9f56aa003c1f922
+Transaction hash: 0xf35c40dbbdc9e4298698ad1cb9937195e5a5e74e557bab1970a5dfd42a32f533
+```
+
+# Using Cast to Interact with the Smart Contract
+We can call functions on our NFT contract with [Cast](https://book.getfoundry.sh/reference/cast/cast-send.html), Foundry's command-line tool for interacting with smart contracts, sending transactions, and getting chain data. In this scenario, we will mint a Game Item to a player's wallet using the [`awardItem` function](https://github.com/ava-labs/avalanche-smart-contract-quickstart/blob/0f29cbb6375a1a452579213f688609c880d52c01/contracts/NFT.sol#L17) in our smart contract.
+
+Given that you already set your RPC and private key env variables during deployment, mint an NFT from your contract by running:
+
+```zsh
+cast send --rpc-url=$RPC_URL <NFT-CONTRACT-ADDRESS> "awardItem(address)" <RECIPIENT-ADDRESS> --private-key=$PRIVATE_KEY
```

Comment:
How to set < NFT-CONTRACT-ADDRESS > and < NFT-CONTRACT-ADDRESS >?

Reply:
How does this sound?

Mint an NFT from your contract by replacing `<NFT-CONTRACT-ADDRESS>` with your `Deployed to` address and `<NFT-RECIPIENT-ADDRESS>` with an address of your choice.
_This section assumes you already set your RPC and private key env variables during deployment_